File: bridge-pattern/python/src/AbstractShape.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

# 定义抽象桥接类，关联图形操作接口
# python用普通类代替亦可
class AbstractShape():
    draw_tool = None

    def __init__(self, draw_tool):
        self.draw_tool = draw_tool

    def draw(x, y, radius):
        pass

File: bridge-pattern/js/src/AbstractShape.js
// 定义抽象形状类，桥接图形操作接口DrawTool
// js没有接口和抽象类，用普通类代替亦可
export class AbstractShape {
  constructor(drawTool) {
    this.drawTool = drawTool
  }

  draw(x, y, radius) {}
}